lib/Frontend/Frontend.cpp

```diff
@@ -745,7 +745,8 @@ void CompilerInstance::parseLibraryFile(
   auto &Diags = NextInput->getASTContext().Diags;
   auto DidSuppressWarnings = Diags.getSuppressWarnings();
   Diags.setSuppressWarnings(DidSuppressWarnings || !IsPrimary);
-
+  if (IsPrimary)
+    NextInput->enableInterfaceHash();
```

Comment:
I think you can simplify this by doing it even earlier, in `CompilerInstance::createSourceFileForMainModule`.

Reply:
I've put it in `recordPrimarySourceFile`.